api/edge_api/identities/mixins.py

```diff
@@ -0,0 +1,93 @@
+import json
+
+import requests
+from django.conf import settings
+from rest_framework.request import Request
+
+from environments.dynamodb import DynamoIdentityWrapper
+from environments.models import Environment
+from util.util import postpone
+
+
+def _should_migrate(project_id: int) -> bool:
+    if not settings.EDGE_API_URL:
+        return False
+    dynamo_identity_wrapper = DynamoIdentityWrapper()
+    return dynamo_identity_wrapper.is_migration_done(project_id)
+
+
+class MigrateIdentitiesUsingRequestsMixin:
+    identities_url = settings.EDGE_API_URL + "identities/"
+
+    def migrate_identity(self, request: Request, environment):
+        self._migrate_identity_async(request, environment)
+
+    @postpone
+    def _migrate_identity_async(self, request: Request, environment: Environment):
+        return self._migrate_identity_sync(request, environment)
+
+    def _migrate_identity_sync(self, request: Request, environment: Environment):
+        if not _should_migrate(environment.project.id):
+            return
+        if request.method == "POST":
+            return self._make_migrate_identity_post_request(
+                request, environment.api_key
+            )
+        self._make_migrate_identity_get_request(request, environment.api_key)
+
+    def _make_migrate_identity_get_request(self, request: Request, api_key: str):
+        return requests.get(
+            self.identities_url,
+            params=request.GET.dict(),
+            headers=_request_headers(api_key),
+        )
+
+    def _make_migrate_identity_post_request(self, request: Request, api_key: str):
+        requests.post(
+            self.identities_url,
+            data=json.dumps(request.data),
```

Comment:
Couldn't we just use `request.body` here?

Reply:
Ah, no, we can only read `request.body` once